wgpu-hal/src/metal/surface.rs

```diff
@@ -208,10 +208,13 @@ impl crate::Surface<super::Api> for super::Surface {
         render_layer.set_pixel_format(self.raw_swapchain_format);
         render_layer.set_framebuffer_only(framebuffer_only);
         render_layer.set_presents_with_transaction(self.present_with_transaction);
+        // opt-in to Metal EDR
+        // EDR potentially more power used in display and more bandwidth, memory footprint.
+        render_layer.set_wants_extended_dynamic_range_content(
+            self.raw_swapchain_format == mtl::MTLPixelFormat::RGBA16Float,
+        );
```

Comment:
Is the only thing I'm not sure about, will defer to the mac experts here.

Reply:
`wantsExtendedDynamicRangeContent` might only available on macOS 10.11+, but tested on iOS and didn't report any error , however setting this true was necessary to be able to display HDR contents

Reply:
There is a concern of calling something that we aren't supposed to call.
Perhaps, we could only call this if the requested format was ever `mtl::MTLPixelFormat::RGBA16Float` for this surface?

Reply:
During the lifetime of a application,  user can switch surface `configure`, and since Metal EDR may consume more power, non-RGBA16Float needs to be set to false

Reply:
Yes, that's why I wrote "if the requested format was ever ". I.e. the surface could keep a flag "HDR property is active", which we'd set once the format is configured to be `RGBA16Float` and keep on. Based on the flag we'd know when to call `set_wants_extended_dynamic_range_content`.

Alternatively, the `PrivateCapabilities` could have a flag for this, saying whether or not we can call `set_wants_extended_dynamic_range_content` at all.

Reply:
I'll modify the code according to your suggestions